### PR TITLE
CO SOS data: races

### PR DESCRIPTION
**Pull request by bhaan**
Adds Colorado Secretary of State General Election race data.

You can test this locally by running the following commands:

```bash
$ cd scrapers/
$ cargo run --bin co_sos_general_candidates
```

The `race` data will be inserted into the database associated with the `DATABASE_URL` environment variable in your `.env` file.

### CODE REVIEW

**Review by wileymc — DISMISSED** [review]
working great locally!

**Comment by wileymc** [comment]
> Adds Colorado Secretary of State General Election race data.
> 
> You can test this locally by running the following commands:
> 
> ```shell
> $ cd scrapers/
> $ cargo run --bin co_sos_general_candidates
> ```
> 
> The `race` data will be inserted into the database associated with the `DATABASE_URL` environment variable in your `.env` file.
> 
> WIP: Needs test, and I think I need to rework the race title formatting.

cc @bastardlee for race title formatting

**Review by wileymc — COMMENTED** [review]
Can you run `cargo clippy` on this and fix some of the warnings (or if you disagree with them add a `clippy.toml` and make your own rules 🦀 

Some of them are from old code which you can ignore (I'll clean that up)

File: scrapers/src/generators/race.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    pub election_scope: &'a db::ElectionScope,
+    pub office_name: Option<&'a str>,
+    pub state: Option<&'a db::State>,
+    pub county: Option<&'a str>,
+    pub district: Option<&'a str>,
+    pub seat: Option<&'a str>,
+    pub year: i32,
+}
+
+impl<'a> RaceTitleGenerator<'a> {
```
Comment (wileymc) [inline scrapers/src/generators/race.rs:20]: Love this setup

**Review by wileymc — DISMISSED** [review]

**Comment by bhaan** [comment]
Ah sorry, didn't realize you were using clippy. All fixed 🫡

**Review by wileymc — APPROVED** [review]

**Pull request merged**